src/main/java/SuperPeer.java

```diff
@@ -21,35 +26,44 @@ public static void main(String[] args) {
     
     public SuperPeer() {
         JavaMeshManager mm = new JavaMeshManager(true);
-        mm.on(TRANSACTION_RECEIVED, this::handleTransactionPacket);
         System.out.println("Superpeer MeshID: " + mm.getUuid());
         
         System.out.println("Waiting for lib to be ready");
         try {
                 Thread.sleep(200);
         } catch(Exception ex) {}
-        
+
+        TransactionsManager tm = TransactionsManager.getInstance(mm);
+        tm.start();
+
         //test send on the TRANSACTION PORT
-        mm.sendDataReliable(mm.getUuid(), MeshUtility.TRANSACTION_PORT, "test".getBytes());
+        //mm.sendDataReliable(mm.getUuid(), MeshUtility.TRANSACTION_PORT, "test".getBytes());
         
         //Continue to run until user quits
         System.out.println("Enter empty line to quit");
-        Console c = System.console();
-        String msg;
-        if (c == null) {
-            System.err.println("No console.");
-            mm.stop();
-            return;
-        }
-        
+        String msg = "";
+//        Console c = System.console();
+//        if (c == null) {
+//            System.err.println("No console.");
+//            mm.stop();
+//            return;
+//        }
+
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+
         do {
-            msg = c.readLine();
+            //msg = c.readLine();
+            try {
+                msg = br.readLine();
+            } catch (IOException e) {
+                e.printStackTrace();
+                break;
+            }
         } while(msg != null && !msg.equals(""));
-        
+
+        tm.stop();
         mm.stop();
+
+        
     }
-    
-    public void handleTransactionPacket(MeshManager.RightMeshEvent e) {
```

Comment:
This shouldn't have been taken out - this is how all of the transaction requests should be handled...this function should parse the request types and execute them in the ethereum network.

Reply:
I moved it to the TransactionManager

Reply:
Yep just saw that - was about to update comment ...haha